### PR TITLE
Split RPG Maker into versions

### LINKED ISSUE #313: OneShot, and presumably other MKXP-based RPG Maker games are not detected
### SteamDB link to the game

https://steamdb.info/app/420530
OneShot is an RPG Maker XP game that uses a fork of the custom open-source RGSS interpreter 'MKXP' for performance, cross-platform support, and interfacing with native libraries.
For information, their fork is located here https://github.com/elizagamedev/mkxp-oneshot

### What should it be detected as?

RPG Maker

### PR DESCRIPTION
**Pull request by MichaelHinrichs**
#### RPG Maker 2000
https://steamdb.info/app/750610/

#### RPG Maker 2003
https://steamdb.info/app/650700/
https://steamdb.info/app/843170/
https://steamdb.info/app/1170880/
https://steamdb.info/app/2572340/

#### RPG Maker XP
https://steamdb.info/app/420530/
https://steamdb.info/app/559210/
https://steamdb.info/app/574740/
https://steamdb.info/app/687980/
https://steamdb.info/app/888070/
https://steamdb.info/app/1098040/
https://steamdb.info/app/1269600/
https://steamdb.info/app/1569440/
https://steamdb.info/app/1836970/
https://steamdb.info/app/2259740/

#### RPG Maker VX
https://steamdb.info/app/461820/
https://steamdb.info/app/526430/
https://steamdb.info/app/1248820/
https://steamdb.info/app/2137560/

#### RPG Maker VX Ace
https://steamdb.info/app/335670/
https://steamdb.info/app/371370/
https://steamdb.info/app/397740/
https://steamdb.info/app/410670/
https://steamdb.info/app/448570/
https://steamdb.info/app/453850/
https://steamdb.info/app/557630/
https://steamdb.info/app/557800/
https://steamdb.info/app/587310/
https://steamdb.info/app/706560/
https://steamdb.info/app/709850/
https://steamdb.info/app/760890/
https://steamdb.info/app/959410/
https://steamdb.info/app/964350/
https://steamdb.info/app/969530/
https://steamdb.info/app/969690/
https://steamdb.info/app/975050/
https://steamdb.info/app/1114650/
https://steamdb.info/app/1120360/
https://steamdb.info/app/1131540/
https://steamdb.info/app/1291100/
https://steamdb.info/app/1313490/
https://steamdb.info/app/1375880/
https://steamdb.info/app/1625080/
https://steamdb.info/app/1693180/
https://steamdb.info/app/1693650/
https://steamdb.info/app/1740100/
https://steamdb.info/app/1788800/
https://steamdb.info/app/1811590/
https://steamdb.info/app/1959510/
https://steamdb.info/app/2250330/
https://steamdb.info/app/2274410/
https://steamdb.info/app/2325150/
https://steamdb.info/app/2491540/

#### RPG Maker MV
https://steamdb.info/app/274310/
https://steamdb.info/app/410670/
https://steamdb.info/app/497090/
https://steamdb.info/app/511630/
https://steamdb.info/app/676800/
https://steamdb.info/app/724520/
https://steamdb.info/app/725980/
https://steamdb.info/app/796910/
https://steamdb.info/app/853250/
https://steamdb.info/app/868320/
https://steamdb.info/app/890690/
https://steamdb.info/app/952510/
https://steamdb.info/app/1035990/
https://steamdb.info/app/1067380/
https://steamdb.info/app/1071900/
https://steamdb.info/app/1094020/
https://steamdb.info/app/1251690/
https://steamdb.info/app/1270570/
https://steamdb.info/app/1286170/
https://steamdb.info/app/1300360/
https://steamdb.info/app/1340730/
https://steamdb.info/app/1345470/
https://steamdb.info/app/1350630/
https://steamdb.info/app/1417970/
https://steamdb.info/app/1432620/
https://steamdb.info/app/1522860/
https://steamdb.info/app/1641210/
https://steamdb.info/app/1834870/
https://steamdb.info/app/1986310/
https://steamdb.info/app/1900810/
https://steamdb.info/app/1902020/
https://steamdb.info/app/1916920/
https://steamdb.info/app/1935570/
https://steamdb.info/app/1971400/
https://steamdb.info/app/2025650/
https://steamdb.info/app/2067780/
https://steamdb.info/app/2105080/
https://steamdb.info/app/2222970/
https://steamdb.info/app/2241200/
https://steamdb.info/app/2277780/
https://steamdb.info/app/2301510/
https://steamdb.info/app/2407010/
https://steamdb.info/app/2578250/

#### RPG Maker MZ
https://steamdb.info/app/1390370/
https://steamdb.info/app/1511270/
https://steamdb.info/app/1704240/
https://steamdb.info/app/1719130/
https://steamdb.info/app/1726280/
https://steamdb.info/app/1823360/
https://steamdb.info/app/1823350/
https://steamdb.info/app/2373160/
https://steamdb.info/app/2469650/
https://steamdb.info/app/2525550/
https://steamdb.info/app/2527350/
https://steamdb.info/app/2543100/

### Brief explanation of the change
Split off each version of RPG Maker into it's own separately detected engine. Discern between versions by version exclusive file types and names. Also, closes #313.

### CODE REVIEW

File: rules.ini [hunk 1]
```diff
@@ -20,6 +20,10 @@ PIXI_JS = (?:^|/)pixi\.js$
 PK3 = \.pk3$ ; idTech3
 RIM = \.rim$ ; BioWare
 RPF = \.rpf$ ; RAGE
+RPGMaker2000[] = (?:^|/)rp[tg]_rt\.exe$
+RPGMaker2000[] = (?:^|/)RPG_RT\.ini$
```
Comment (Lifeismana) [inline rules.ini:24]: you should rename these 2 evidences to `RPGMaker200x` if they are for for both 2000 & 2003

Comment (MichaelHinrichs) [reply rules.ini:24]: Oh, good idea.

Comment (MichaelHinrichs) [reply rules.ini]: That's weird, i just made the changes, but some of the errors that are coming up still mention RPG Maker 2000.

Comment (MichaelHinrichs) [reply rules.ini]: Oh wait, i think they still say that because "FileDetector.php" decides that the engine is RPG Maker 2000 if it doesn't look like 2003.

**Review by Lifeismana — CHANGES_REQUESTED** [review]
These descriptions need to be less opinionated and more neutral

**Comment by Lifeismana** [comment]
In the end, tracking per version engines is not something we are interested in

**Comment by gBasil** [comment]
> In the end, tracking per version engines is not something we are interested in

Do you mean that in general, or just for RPG Maker?

The reason I ask is because the RPG Maker series treats each major software (re)release (MZ, MV, VX, 20003, etc.) as a different product; each version has it's own versioning system (I might be off, but MZ is currently on 1.8.0, MV seems to be on 1.6.2, and 2003 at least reached 1.12), and sometimes use a completely different platform (i.e. going from being Ruby-based to being web-based to being Unity-based).

Engines like Godot and Unity just bump their version number but are still referred to by the same name. I'm tempted to compare the RPG Maker software more with the Source engine and how it's evolved (i.e. GoldSrc, Source, Source 2), which all seem to have their own distinct entries here:

- https://github.com/SteamDatabase/FileDetectionRuleSets/blob/main/tests/types/Engine.GoldSource.txt
- https://github.com/SteamDatabase/FileDetectionRuleSets/blob/main/tests/types/Engine.Source.txt
- https://github.com/SteamDatabase/FileDetectionRuleSets/blob/main/tests/types/Engine.Source2.txt

I'm just wondering what the thought process was, since it wasn't outlined in the comment.

**Pull request closed without merging**